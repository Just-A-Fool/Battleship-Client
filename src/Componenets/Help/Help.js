import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Help.css';

export default class Help extends Component {

  render() {
    return (

    );
  };
  
};


