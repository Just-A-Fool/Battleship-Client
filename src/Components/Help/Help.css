.helpIB ul {
  list-style-type: none;
  display: inline;
}

.help {
  margin-top: 8rem;
}

.help h1 {
  margin-bottom: 2rem;
}

li {
  font-size: 1.3rem;
  display: block;
}

.helpIB {
  border: 0.1rem solid white;
  padding-bottom: 2.2rem;
  margin-bottom: 2rem !important;
  width: 90%;
  margin: 0 auto;
  border-radius: 1rem;
  background-color: rgb(30, 52, 90);
}

@media screen and (min-width: 900px) {
  .helpIB {
    width: 80%;
  }
}

@media screen and (min-width: 1025px) {
  .helpIB {
    width: 65%;
  }
}

