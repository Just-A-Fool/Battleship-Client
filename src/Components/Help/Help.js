import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import Button from '../Button/Button';
import './Help.css';

class Help extends Component {

  render() {
    return (
      <div className='help'>
        <h1>Help</h1>
        <div>
          <h2>Instructions</h2>
          <ul>
            <li>1. How to Play</li>
            <li>2. How to Play</li>
            <li>3. How to Play</li>
            <li>4. How to Play</li>
            <li>5. How to Play</li>
          </ul>
        </div>

        <div>
          <h2>Bug Report</h2>
          <ul>
            <li>1. Contact Info</li>
            <li>2. Contact Info</li>
            <li>3. Contact Info</li>
            <li>4. Contact Info</li>
          </ul>
        </div>

        <Button>
          <Link to='/dashboard'>
            Exit
          </Link>
        </Button>

        <footer>
          Copyright Â© since 2020
        </footer>
      </div>
    );
  };
  
};

export default withRouter(Help);
