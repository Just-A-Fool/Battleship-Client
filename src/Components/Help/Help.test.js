import React from 'react';
import ReactDOM from 'react-dom';
import Help from './Help';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';


describe('Help component', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><Help /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Displays the Help component when rendered', () => {
    const wrapper = mount(<Router><Help /></Router>)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it(`Contains an 'h3' element for showing how to play`, () => {
    const wrapper = mount(<Router><Help /></Router>)
    expect(wrapper.contains(<h3>How to play</h3>)).toEqual(true)
  });

  it(`Contains an 'h3' element for showing how to resume an active game`, () => {
    const wrapper = mount(<Router><Help /></Router>)
    expect(wrapper.contains(<h3>Resuming an active game</h3>)).toEqual(true)
  });

  it(`Contains an 'h3' element for showing how to start a new game`, () => {
    const wrapper = mount(<Router><Help /></Router>)
    expect(wrapper.contains(<h3>New Game</h3>)).toEqual(true)
  });
  
});