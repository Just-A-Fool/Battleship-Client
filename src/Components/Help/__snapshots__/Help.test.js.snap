// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help component Displays the Help component when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Help)>
      <Route>
        <Help
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div>
            <withRouter(Header)>
              <Route>
                <Header
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <div>
                    <Banner>
                      <div
                        className="banner"
                      >
                        <h1>
                          BATTLESHIP
                        </h1>
                      </div>
                    </Banner>
                    <header
                      className="main-header"
                    >
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Dashboard
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                           Help
                        </button>
                      </ForwardRef>
                      <ForwardRef
                        onClick={[Function]}
                      >
                        <button
                          className="Button"
                          onClick={[Function]}
                        >
                          Logout
                        </button>
                      </ForwardRef>
                    </header>
                  </div>
                </Header>
              </Route>
            </withRouter(Header)>
            <div
              className="help"
            >
              <h1>
                Help
              </h1>
              <div
                className="helpIB"
              >
                <h2>
                  Instructions
                </h2>
                <p>
                  Key:
                </p>
                <div
                  className="key"
                >
                  <div>
                    <div
                      className="cell"
                    />
                    <p>
                      Open Target
                    </p>
                  </div>
                  <div>
                    <div
                      className="cell selected"
                    />
                    <p>
                      Selected Target
                    </p>
                  </div>
                  <div>
                    <div
                      className="cell hit"
                    />
                    <p>
                      Hit Target
                    </p>
                  </div>
                  <div>
                    <div
                      className="cell miss"
                    />
                    <p>
                      Missed Target
                    </p>
                  </div>
                  <div>
                    <div
                      className="cell ship"
                    />
                    <p>
                      Ship Location
                    </p>
                  </div>
                  <div>
                    <div
                      className="cell sunk"
                    />
                    <p>
                      Sunken Ship
                    </p>
                  </div>
                </div>
                <h3>
                  How to play
                </h3>
                <ol>
                  <li>
                    1. Once both users have set their ships the match starts
                  </li>
                  <li>
                    2. Players will take turns 'firing' upon eachother's ships attempting to 'sink' them
                  </li>
                  <li>
                    3. Once one player has 'sunk' all of their opponent's ships the player wins
                  </li>
                </ol>
                <h3>
                  New Game
                </h3>
                <ol>
                  <li>
                    1. If you want to start a new game, then select "Start a New Game". 
                  </li>
                  <li>
                    2. Place each of your five ships one at a time on the grid.
                  </li>
                  <ul>
                    <li>
                      To do so select an origin point. This is done by either clicking a tile on the grid or using the dropdown selectors
                    </li>
                    <li>
                      Also set an alignment through use of the button labeled either 'Horizontal' or 'Vertical'. By clicking this button you may cycle between the two alignments.  The alignment shown is the chosen alignment.
                    </li>
                    <li>
                      Once the green outline matches where you wish to place your ship you may click the submit button to set the ship
                    </li>
                  </ul>
                  <li>
                    3. Once both players have set their ships the match begins.
                  </li>
                </ol>
                <h3>
                  Resuming an active game
                </h3>
                <ol>
                  <li>
                    1. To resume a game its as easy as clicking the 'Resume Game' button on the game you wish to enter.
                  </li>
                  <li>
                    2. Take note that not all games will have an opponent to start off with as we wait to match you against another person.
                  </li>
                  <li>
                    3. Games where it is your turn to move are nearer to the top.
                  </li>
                </ol>
                <h3>
                  Forfeiting a match
                </h3>
                <ol>
                  <li>
                    1. If you no longer wish to continue a match you may select the 'Quit Game' button then the 'Quit Game Now' button from the dashboard.
                  </li>
                </ol>
              </div>
              <ForwardRef
                onClick={[Function]}
              >
                <button
                  className="Button"
                  onClick={[Function]}
                >
                   Back
                </button>
              </ForwardRef>
              <Footer>
                <footer>
                  Copyright Â© since 2020
                </footer>
              </Footer>
            </div>
          </div>
        </Help>
      </Route>
    </withRouter(Help)>
  </Router>
</BrowserRouter>
`;
